src/image/channel_groups.rs

```diff
@@ -155,43 +205,42 @@ impl<'slf, ChannelGroup> WritableChannels<'slf> for ChannelGroups<ChannelGroup>
         mode
     }
 
-    type Writer = GroupChannelsWriter<'slf, ChannelGroup>;
+    type Writer = GroupChannelsWriter<ChannelGroup::Writer>;
 
     fn create_writer(&'slf self, header: &Header) -> Self::Writer {
-        let channels = header.channels.list.iter()
-            .map(|channel_info|{
-                // hashmap order is not guaranteed? so look up each channel group manually instead of generating new
-                let channels = self.lookup_group(channel_info.name.as_slice())
-                    .expect("channels not found bug");
-
-                channels.create_writer(header) // channel_info.name.clone()
-            })
-            .collect();
-
-        GroupChannelsWriter { channels_list: channels }
+        GroupChannelsWriter {
+            all_channel_groups: self.all_channel_groups()
+                .map(|channel_group: &ChannelGroup| panic!("this uses relative names but expects absolute names, and all will write first byte")/*channel_group.create_channel_group_writer(header)*/)
```

Comment:
`SpecificChannels` currently does a lookup based on the channel name, in order to find the byte offset, for each channel. This cannot work as the meta data channels have absolute names while the `SpecificChannels` don't